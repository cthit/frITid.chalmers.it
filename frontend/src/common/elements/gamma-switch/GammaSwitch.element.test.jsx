import React from "react";
import { shallow } from "enzyme";
import GammaSwitch from "./";

describe("<GammaSwitch/>", () => {
  test("Shallow render of primary <GammaSwitch/>", () => {
    const wrapper = shallow(
      <GammaSwitch value={false} primary label="1" onChange={e => {}} />
    );

    expect(wrapper).toMatchSnapshot();
  });

  test("Shallow render of secondary <GammaSwitch/>", () => {
    const wrapper = shallow(
      <GammaSwitch value={true} secondary label="2" onChange={e => {}} />
    );

    expect(wrapper).toMatchSnapshot();
  });

  test("Shallow render of <GammaSwitch/> with a name", () => {
    const wrapper = shallow(
      <GammaSwitch
        value={false}
        label="3"
        name="Glass"
        onChange={e => {}}
        onBlur={e => {}}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  test("Shallow render of a normal <GammaSwitch/> with an error", () => {
    const wrapper = shallow(
      <GammaSwitch value={true} label="4" error={true} onChange={e => {}} />
    );

    expect(wrapper).toMatchSnapshot();
  });

  test("Shallow render of a disabled <GammaSwitch/>", () => {
    const wrapper = shallow(
      <GammaSwitch value={false} disabled label="5" onChange={e => {}} />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
