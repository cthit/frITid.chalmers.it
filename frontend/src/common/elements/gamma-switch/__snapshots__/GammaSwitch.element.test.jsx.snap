// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GammaSwitch/> Shallow render of <GammaSwitch/> with a name 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GammaSwitch
    label="3"
    name="Glass"
    onBlur={[Function]}
    onChange={[Function]}
    value={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "control": <WithStyles(Switch)
        checked={false}
        color="default"
        disabled={undefined}
        name="Glass"
        onBlur={[Function]}
        onChange={[Function]}
/>,
      "error": undefined,
      "label": "3",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "control": <WithStyles(Switch)
          checked={false}
          color="default"
          disabled={undefined}
          name="Glass"
          onBlur={[Function]}
          onChange={[Function]}
/>,
        "error": undefined,
        "label": "3",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<GammaSwitch/> Shallow render of a disabled <GammaSwitch/> 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GammaSwitch
    disabled={true}
    label="5"
    onChange={[Function]}
    value={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "control": <WithStyles(Switch)
        checked={false}
        color="default"
        disabled={true}
        name={undefined}
        onBlur={undefined}
        onChange={[Function]}
/>,
      "error": undefined,
      "label": "5",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "control": <WithStyles(Switch)
          checked={false}
          color="default"
          disabled={true}
          name={undefined}
          onBlur={undefined}
          onChange={[Function]}
/>,
        "error": undefined,
        "label": "5",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<GammaSwitch/> Shallow render of a normal <GammaSwitch/> with an error 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GammaSwitch
    error={true}
    label="4"
    onChange={[Function]}
    value={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "control": <WithStyles(Switch)
        checked={true}
        color="default"
        disabled={undefined}
        name={undefined}
        onBlur={undefined}
        onChange={[Function]}
/>,
      "error": true,
      "label": "4",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "control": <WithStyles(Switch)
          checked={true}
          color="default"
          disabled={undefined}
          name={undefined}
          onBlur={undefined}
          onChange={[Function]}
/>,
        "error": true,
        "label": "4",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<GammaSwitch/> Shallow render of primary <GammaSwitch/> 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GammaSwitch
    label="1"
    onChange={[Function]}
    primary={true}
    value={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "control": <WithStyles(Switch)
        checked={false}
        color="primary"
        disabled={undefined}
        name={undefined}
        onBlur={undefined}
        onChange={[Function]}
/>,
      "error": undefined,
      "label": "1",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "control": <WithStyles(Switch)
          checked={false}
          color="primary"
          disabled={undefined}
          name={undefined}
          onBlur={undefined}
          onChange={[Function]}
/>,
        "error": undefined,
        "label": "1",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<GammaSwitch/> Shallow render of secondary <GammaSwitch/> 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GammaSwitch
    label="2"
    onChange={[Function]}
    secondary={true}
    value={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "control": <WithStyles(Switch)
        checked={true}
        color="secondary"
        disabled={undefined}
        name={undefined}
        onBlur={undefined}
        onChange={[Function]}
/>,
      "error": undefined,
      "label": "2",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "control": <WithStyles(Switch)
          checked={true}
          color="secondary"
          disabled={undefined}
          name={undefined}
          onBlur={undefined}
          onChange={[Function]}
/>,
        "error": undefined,
        "label": "2",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
